web/source/text/kbdInterface.ts

```diff
@@ -913,21 +917,24 @@ namespace com.keyman.text {
      * value across sessions, providing a custom user default for later uses of the keyboard.
      */    
     saveStore(storeName:string, optValue:string): boolean {
-      let keyman = com.keyman.singleton;
       this.resetContextCache();
       var kbd=this.activeKeyboard;
       if(!kbd || typeof kbd.id == 'undefined' || kbd.id == '') {
         return false;
       }
-      
-      // The cookie entry includes the store name...
-      var cName='KeymanWeb_'+kbd.id+'_Option_'+storeName;
-      var cValue=encodeURIComponent(optValue);
 
       // And the lookup under that entry looks for the value under the store name, again.
-      let valueObj = {};
-      valueObj[storeName] = cValue;
-      keyman.util.saveCookie(cName, valueObj);
+      let valueObj: VariableStore = {};
+      valueObj[storeName] = optValue;
+
+      // Null-check in case of invocation during unit-test
+      if(this.ruleBehavior) {
+        this.ruleBehavior.saveStore[storeName] = valueObj;
+      } else {
+        // We're in a unit-test environment, directly invoking this method from outside of a keyboard.
+        // In this case, we should immediately commit the change.
+        this.variableStoreSerializer.saveStore(this.activeKeyboard.id, storeName, this.saveStore[storeName]);
```

Comment:
Doing this is a bit ugly, but the alternative would require higher-maintenance references on related unit tests.  

`saveStore`'s a published API and isn't changing, while the internal implementation (with the new `variableStoreSerializer`) is quite new and may need further tweaks.